Classes/PrunnerApiService.php

```diff
@@ -54,13 +54,19 @@ public function loadPipelinesAndJobs(): PipelinesAndJobsResponse
     {
         $resultString = $this->apiCall('GET', 'pipelines/jobs', null)->getBody()->getContents();
         $result = json_decode($resultString, true);
+        if ($result === null) {
+            throw new \RuntimeException('Could not decode JSON response from prunner API');
+        }
         return PipelinesAndJobsResponse::fromJsonArray($result);
     }
 
     public function loadJobDetail(JobId $jobId): ?Job
     {
         $resultString = $this->apiCall('GET', 'job/detail?' . http_build_query(['id' => $jobId->getId()]), null)->getBody()->getContents();
         $result = json_decode($resultString, true);
+        if ($result === null) {
+            throw new \RuntimeException('Could not decode JSON response from prunner API');
+        }
         if (isset($result['error'])) {
             return null;
         }
```

Comment:
shouldnt we prefer `JSON_THROW_ON_ERROR` for a more meaning full exception?

Reply:
Will test if it results in the same

Reply:
Adjusted it but wrapped the exception into another one to make it better understandable.